SalesforceCommerceCloud/pwa-kit
# @W-16778724 Disable `usePage` react hook if `shopperContext` is undefined in `PageDesignerPromotionalBanner`

# Description



To reduce unnecessary logs and API calls, when there is no shopper context set we will not send a GET request to the `/pages` API in `PageDesignerPromotionalBanner`.

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- disable the `usePage` hook when shopper context is not set in `PageDesignerPromotionalBanner`

# How to Test-Drive This PR

This branches code has been deployed in https://runtime-admin-jinsu.mobify-storefront.com/salesforce-internal/cloud-jinsu/env2

- Start the local server using `npm start`
- Navigate to the PLP page: https://cloud-jinsu-env2.mobify-storefront-staging.com/global/en-GB/category/womens-clothing-tops
- Open the developer console and switch to the *Network* tab
- Verify that the `/shopper-context` API returns a 404 and that there are no `/shopper-experience` API calls
- Set the shopper context to `sourceCode=fb`: https://cloud-jinsu-env2.mobify-storefront-staging.com/global/en-GB/category/womens-clothing-tops?sourceCode=fb
- Verify 2 `/shopper-experience` API calls are made but return a 500
  - `/shopper-experience/v1/organizations/f_ecom_zzrf_001/pages/instagram-promo-banner-desktop?siteId=RefArchGlobal&locale=en-GB`
  - `shopper-experience/v1/organizations/f_ecom_zzrf_001/pages/instagram-promo-banner-desktop?siteId=RefArchGlobal&locale=en-GB`
- Set the shopper context to `sourceCode=instagram`: https://cloud-jinsu-env2.mobify-storefront-staging.com/global/en-GB/category/womens-clothing-tops?sourceCode=instagram
- Verify the 2 `/shopper-experience` API calls are made and return a 200
- Verify the banner is displayed

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
